jupyterlab_translate/utils.py

```diff
@@ -289,13 +291,65 @@ def get_line(lines: List[str], value: str) -> str:
     return str(line_count)
 
 
+def _extract_schema_strings(
+    schema: dict, schema_lines: list, ref_path: str, prefix: str = ""
+):
+    message_context = "schema"
+
+    to_translate = [
+        "title",
+        "description",
+        "properties/.*/title",
+        "properties/.*/description",
+        "definitions/properties/.*/title",
+        "definitions/properties/.*/description",
+        # JupyterLab-specific
+        "jupyter.lab.setting-icon-label",
+        "jupyter.lab.menus/.*/label"
+        # TODO: toolbar?
+        # TODO: add custom paths specified in jupyter.lab.internationalization
+    ]
+    to_translate = [re.compile("^/" + pattern + "$") for pattern in to_translate]
+
+    entries = []
+
+    for key, value in schema.items():
+        path = prefix + "/" + key
+
+        if isinstance(value, str):
+            if any([pattern.fullmatch(path) is not None for pattern in to_translate]):
+                text = value.replace("\n", "</br/>")
+                entries.append(
+                    dict(
+                        # TODO should we add the path to context? It seems that it could be useful for translators,
+                        #      but would this be backward-compatible with existing translations?
```

Comment:
Any thoughts on this one @goanpeca? By path I mean the e.g. `/definitions/properties/cursorBlinkRate/title/`

Reply:
Or maybe it should go to occurrences instead?

Reply:
Are the occurrences formalized in any way?

Reply:
I do not follow @krassowski

Reply:
I would add the path as context.